packages/app/src/cli/services/app-logs/poll-app-logs.ts

```diff
@@ -41,15 +42,31 @@ export const pollAppLogs = async ({
     },
   })
 
-  if (response.status === 401) {
-    await resubscribeCallback()
-    return
-  }
-
   if (!response.ok) {
-    // We should add some exponential backoff here to not spam partners
-
     const responseText = await response.text()
+    if (response.status === 401) {
+      await resubscribeCallback()
+      return
+    } else if (response.status === 429 || response.status >= 500) {
+      stdout.write(`Received an error while polling for app logs.`)
+      stdout.write(`${response.status}: ${response.statusText}`)
+      stdout.write(responseText)
+      stdout.write(`Retrying in 10 seconds`)
+      setTimeout(() => {
+        pollAppLogs({
+          stdout,
+          appLogsFetchInput: {
+            jwtToken,
+            cursor: undefined,
+          },
+          apiKey,
+          resubscribeCallback,
+        }).catch((error) => {
+          throw new Error(`${error} error while fetching.`)
+        })
```

Comment:
Added this because of the lint rule, but it feels odd. There's a few different places now that all have similar catch/re-throw lines. Any thoughts on a way to clean this up while still keeping the linter happy?

Reply:
What was the lint violation?